File: samples/redis/HelloSequence.cs
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the project root for license information.

using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.Azure.WebJobs;

namespace VSSample
{
    public static class HelloSequence
    {
        [FunctionName("HelloSequence")]
        public static async Task<List<string>> Run(
            [OrchestrationTrigger] IDurableOrchestrationContext context)
        {
            var outputs = new List<string>();

            outputs.Add(await context.CallActivityAsync<string>("SayHello", "Tokyo"));
            outputs.Add(await context.CallActivityAsync<string>("SayHello", "Seattle"));
            outputs.Add(await context.CallActivityAsync<string>("SayHello", "London"));

            // returns ["Hello Tokyo!", "Hello Seattle!", "Hello London!"]
            return outputs;
        }

        [FunctionName("SayHello")]
        public static string SayHello([ActivityTrigger] string name)
        {
            return $"Hello {name}!";
        }
    }
 }

File: samples/fsharp/HelloSequence.fs
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the project root for license information.

namespace VSSample

open Microsoft.Azure.WebJobs
open Microsoft.Azure.WebJobs.Extensions.DurableTask
open FSharp.Control.Tasks

module HelloSequence =

  [<FunctionName("E1_HelloSequence")>]
  let Run([<OrchestrationTrigger>] context: IDurableOrchestrationContext) = task {
    let! hello1 = context.CallActivityAsync<string>("E1_SayHello", "Tokyo")
    let! hello2 = context.CallActivityAsync<string>("E1_SayHello", "Seattle")
    let! hello3 = context.CallActivityAsync<string>("E1_SayHello", "London")

    // returns ["Hello Tokyo!", "Hello Seattle!", "Hello London!"]
    return [hello1; hello2; hello3]
  }       

  [<FunctionName("E1_SayHello")>]
  let SayHello([<ActivityTrigger>] name) =
    sprintf "Hello %s!" name

